### PR TITLE
e2e: fix flakiness in dynamic-pools test03-rebalancing

### PR DESCRIPTION
**Pull request by askervin**
Optimize verifications, make first verifications quickly without long stabilizing periods, retry if needed.

### CODE REVIEW

File: test/e2e/policies.test-suite/dynamic-pools/n4c16/test03-rebalancing/code.var.sh [hunk 1]
```diff
@@ -36,41 +48,44 @@ verify-metrics-has-line 'containers="pod2:pod2c0".*dynamicPool="pool2",dynamicPo
...
 vm-command "nohup kubectl exec pod2 -- /bin/sh -c 'gzip </dev/zero >/dev/null' </dev/null >&/dev/null &"
 # Wait at least one rebalancing round and print CPU pinning.
-sleep 10
+sleep 2
 report allowed
 # Verify that the number of CPUs in pool2 is the largest.
-verify 'len(cpus["pod2c0"]) > len(cpus["pod0c0"])'
-verify 'len(cpus["pod2c0"]) > len(cpus["pod1c0"])'
-verify 'len(cpus["pod0c0"]) + len(cpus["pod1c0"]) + len(cpus["pod2c0"]) == 14'
+retry 10 verify \
```
Comment (klihub) [inline test/e2e/policies.test-suite/dynamic-pools/n4c16/test03-rebalancing/code.var.sh:67]: @askervin @marquiz #1067 adds an implicit unparametrized builtin retry to `verify`. We need to chose whether we go with that, and make it configurable if needed, or with this.

Comment (askervin) [reply test/e2e/policies.test-suite/dynamic-pools/n4c16/test03-rebalancing/code.var.sh:67]: I'd rather change this into `verify --retry 10 ...` on top of PR #1069.

Comment (klihub) [reply test/e2e/policies.test-suite/dynamic-pools/n4c16/test03-rebalancing/code.var.sh:67]: I agree and suggested an alternative in #1067

Comment (marquiz) [reply test/e2e/policies.test-suite/dynamic-pools/n4c16/test03-rebalancing/code.var.sh:67]: > I'd rather change this into `verify --retry 10 ...` on top of PR #1069.

OK, if we need 10 retries here then maybe #1069 would be the way to go

**Comment by askervin** [comment]
Closing this PR, because PR #1069 brings in automatic retries in verify.

**Comment by marquiz** [comment]
> Closing this PR, because PR #1069 brings in automatic retries in verify.

Is the three times from there enough? We can wait an see if any issues regarding this arise

**Pull request closed without merging**